p2p/discover/database.go

```diff
@@ -226,15 +226,15 @@ func (db *nodeDB) ensureExpirer() {
 // expirer should be started in a go routine, and is responsible for looping ad
 // infinitum and dropping stale data from the database.
 func (db *nodeDB) expirer() {
-	tick := time.Tick(nodeDBCleanupCycle)
+	tick := time.NewTicker(nodeDBCleanupCycle)
 	for {
 		select {
-		case <-tick:
+		case <-tick.C:
 			if err := db.expireNodes(); err != nil {
 				log.Error("Failed to expire nodedb items", "err", err)
 			}
-
 		case <-db.quit:
+			tick.Stop()
```

Comment:
It's cleaner to do a `defer tick.Stop()` after creating the ticker. That way the code is simpler and more robust in the face of future changes / multiple return paths.

Reply:
Right, thank you, pushed.